src/Data/Managers/GetCompilationState/QueryManager.ts

```diff
@@ -0,0 +1,67 @@
+import { useContext, useEffect, useState } from "react";
+import {
+  ApiHelper,
+  ContinuousQueryManager,
+  QueryManagerKind,
+  Query,
+  RemoteData,
+  Scheduler,
+} from "@/Core";
+import { DependencyContext } from "@/UI/Dependency";
+
+export class GetCompilationStateQueryManager
+  implements ContinuousQueryManager<"GetCompilationState">
+{
+  constructor(
+    private readonly apiHelper: ApiHelper,
+    private readonly scheduler: Scheduler
+  ) {}
+
+  useContinuous(
+    query: Query.SubQuery<"GetCompilationState">
+  ): [RemoteData.RemoteData<undefined, boolean>, () => void] {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Why do you need to disable the linting rule specifically?

Reply:
But, there is I suppose a good reason why that shouldn't be used, right? I remember it being a hard rule for a reason, I checked an I saw it was used in multiple places, but I have a hunch that the guys before us did bypass the rule just because they were lazy to understand it.